pupil-spa/src/app/services/check-complete/check-complete.service.ts

```diff
@@ -0,0 +1,111 @@
+import { Injectable } from '@angular/core';
+import { APP_CONFIG } from '../config/config.service';
+import {
+  CheckSubmissionApiCalled,
+  CheckSubmissionAPIFailed,
+  CheckSubmissionAPICallSucceeded,
+} from '../audit/auditEntry';
+import { AzureQueueService } from '../azure-queue/azure-queue.service';
+import { AuditService } from '../audit/audit.service';
+import { SubmissionService } from '../submission/submission.service';
+import { StorageService } from '../storage/storage.service';
+import { TokenService } from '../token/token.service';
+import { Router } from '@angular/router';
+
+/**
+ * Declaration of check start service
+ */
+@Injectable()
+export class CheckCompleteService {
+
+  featureUseHpa;
+  checkSubmissionApiErrorDelay;
+  checkSubmissionAPIErrorMaxAttempts;
+  submissionPendingViewMinDisplay;
+
+  constructor(private azureQueueService: AzureQueueService,
+              private submissionService: SubmissionService,
+              private storageService: StorageService,
+              private tokenService: TokenService,
+              private auditService: AuditService,
+              private router: Router) {
+    const { featureUseHpa,
+      checkSubmissionApiErrorDelay,
+      checkSubmissionAPIErrorMaxAttempts,
+      submissionPendingViewMinDisplay,
+    } = APP_CONFIG;
+    this.featureUseHpa = featureUseHpa;
+    this.checkSubmissionApiErrorDelay = checkSubmissionApiErrorDelay;
+    this.checkSubmissionAPIErrorMaxAttempts = checkSubmissionAPIErrorMaxAttempts;
+    this.submissionPendingViewMinDisplay = submissionPendingViewMinDisplay;
+  }
+
+  private sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  /**
+   * Check complete submission
+   * @param {Number} startTime
+   * @returns {Promise.<void>}
+   */
+  public async submit(startTime): Promise<void> {
+    if (this.featureUseHpa === true) {
+      const queueName = 'check-complete';
+      const { url, token } = this.tokenService.getToken('checkComplete');
+      const payload = this.storageService.getAllItems();
+      const retryConfig = {
+        errorDelay: this.checkSubmissionApiErrorDelay,
+        errorMaxAttempts: this.checkSubmissionAPIErrorMaxAttempts
+      };
+      try {
+        this.auditService.addEntry(new CheckSubmissionApiCalled());
+        await this.azureQueueService.addMessage(queueName, url, token, payload, retryConfig);
+        this.auditService.addEntry(new CheckSubmissionAPICallSucceeded());
+        await this.onSuccess(startTime);
+      } catch (error) {
+        const auditEntry = new CheckSubmissionAPIFailed(error);
+        await this.onError(auditEntry);
+      }
+    } else {
+      try {
+        await this.submissionService.submitData().toPromise();
+        this.auditService.addEntry(new CheckSubmissionAPICallSucceeded());
+        this.auditService.addEntry(new CheckSubmissionApiCalled());
```

Comment:
`CheckSubmissionApiCalled()` should be added before the submission and `CheckSubmissionAPICallSucceeded()` after it.

then in the catch block call a similar `...failed()` method

Reply:
@GuyHarwood I changed the order between called and succeeded however failed is not audited as part of this service as the observable within submission service (part of the existing implementation) will emit it for a maximum of `checkSubmissionAPIErrorMaxAttempts`. Therefore check complete service does not need to audit it in addition.